assets/ts/schedule/components/SchedulePage.tsx

```diff
@@ -0,0 +1,446 @@
+          directionChanged={changeDirection}
+          initialMode={modalMode}
+          initialDirection={directionId}
+          initialOrigin={selectedOrigin}
+          handleOriginSelectClick={handleOriginSelectClick}
+          originChanged={changeOrigin}
+          route={route}
+          routePatternsByDirection={routePatternsByDirection}
+          services={services}
+          stops={stops}
+          today={today}
+          updateURL={updateURL}
+          scheduleNote={scheduleNote}
+        />
+      )}
+    </>
+  );
+};
+
+const getAdditionalLineInfo = (
+  schedulePageData: SchedulePageData,
+  routeIsSuspended: boolean
+): JSX.Element => {
+  const {
+    route,
+    schedule_note: scheduleNote,
+    teasers,
+    pdfs,
+    connections,
+    fares,
+    fare_link: fareLink,
+    holidays,
+    hours
+  } = schedulePageData;
+
+  if (routeIsSuspended) {
+    return (
+      <>
+        <ContentTeasers teasers={teasers} />
+        <UpcomingHolidays holidays={holidays} />
+      </>
+    );
+  }
+
+  return (
+    <AdditionalLineInfo
+      teasers={teasers}
+      pdfs={pdfs}
+      connections={connections}
+      fares={fares}
+      fareLink={fareLink}
+      route={route}
+      holidays={holidays}
+      hours={hours}
+      scheduleNote={scheduleNote}
+    />
+  );
+};
+
+const getPageTitle = (route: Route): string | null => {
+  switch (route.type) {
+    case 0:
+    case 1:
+      return "Stations & Departures";
+    case 4:
+      return null;
+    case 2:
+    default:
+      return "Schedules & Maps";
+  }
+};
+
+interface SchedulePageProps {
+  schedulePageData: SchedulePageData;
+  noBranches: boolean;
+  mapData: MapData;
+  staticMapData?: StaticMapData;
+}
+
+export const SchedulePage = ({
+  schedulePageData,
+  noBranches,
+  mapData,
+  staticMapData
+}: SchedulePageProps): JSX.Element => {
+  const dispatch = useDispatch();
+  const { route, route_patterns: routePatternsByDirection } = schedulePageData;
+
+  const routeIsSuspended = Object.keys(routePatternsByDirection).length === 0;
+
+  const [query] = useQueryParams({
+    // eslint-disable-next-line camelcase
+    "schedule_finder[direction_id]": StringParam,
+    "schedule_direction[direction_id]": StringParam,
+    "schedule_finder[origin]": StringParam
+  });
+
+  const currentState = useSelector((state: StoreProps) => state);
+
+  useEffect(() => {
+    // get initial values from the store:
+    const { selectedDirection, selectedOrigin } = currentState;
+    let { modalOpen, modalMode } = currentState;
+
+    let newDirection: DirectionId | undefined;
+    const newOrigin: SelectedOrigin | undefined =
+      query["schedule_finder[origin]"];
+
+    // modify the store values in case URL has parameters:
+    const queryDirectionId =
+      query["schedule_finder[direction_id]"] ||
+      query["schedule_direction[direction_id]"];
+    if (queryDirectionId !== undefined) {
+      newDirection = queryDirectionId === "0" ? 0 : 1;
```

Comment:
**suggestion**: I believe `queryDirectionId.toString()` will work in this case? will also let us have one fewer code branch to test. :)

Reply:
newDirectionId is only allowed to be the number 0 or 1 so this stops users from entering 76, and also converts the string to s number